flow2ml/Data_Augumentation.py

```diff
@@ -497,4 +497,66 @@ def applyThreshold(self,classPath):
                 # saving the image by
                 plt.imsave(classPath+"/ThresholdedImages/s_threshed"+image,s_threshed)
               except Exception as e:
-                print(f"simple thresholding operation failed due to {e}")
\ No newline at end of file
+                print(f"simple thresholding operation failed due to {e}")
+
+  def changeColorSpace(self,classPath):
+    ''' 
+      Applies changing color-space augmentation to all the images in the given folder.
+      Args : 
+        classPath : (string) directory containing images for a particular class.
+    '''
+    try:
+      os.mkdir(classPath+"/ChangedImages")    
+    except:
+      raise Exception("Unable to create directory for changed images.")
+    for image in list(os.listdir(classPath)):
+
+      if((self.operations['color-space']['input'] not in ['BGR','GRAY']) or (self.operations['color-space']['output'] not in ['BGR','RGB','GRAY'])):
+        raise Exception("Invalid color-space operation.")
+      else:
+
+        if(self.operations['color-space']['input']=='BGR'):
+
+          #Read image in BGR mode.
+          img=cv2.imread(classPath+"/"+image)
+          if img is not None:
+            if (self.operations['color-space']['output']=='RGB'):
+              try:
+                # changing color-space of the image from BGR to RGB.
+                changed = cv2.cvtColor(img,cv2.COLOR_BGR2RGB)  
+                # saving the image
+                plt.imsave(classPath+"/ChangedImages/Changed"+image, changed)
+              except Exception as e:
+                print(f"color-space operation failed due to {e}")
+
+            if (self.operations['color-space']['output']=='GRAY'):
```

Comment:
Code for Converting to greyscale is already in place as apart of Augmentations. Hence, please remove this

Reply:
okay.